app/models/grda_warehouse/auth_policies/source_client_policy.rb

```diff
@@ -83,7 +85,19 @@ def add_legacy_data_source_permissions(results)
     # is the client in a window data source?
     return unless context.legacy_window_data_source_ids.include?(client.data_source_id)
 
-    # check roi if the window config requires release (the "can_*_with_roi" permissions are not relevant in this case)
+    # The 'can_view_clients' and 'can_search_all_clients' permissions allow the user's permissions to apply to clients in windowed data sources, regardless of ROI.
+    # For example a user with can_view_clients and can_view_name can view the names of all clients in windowed data sources
```

Comment:
```suggestion
    # The 'can_view_clients' and 'can_search_all_clients' permissions allow certain permissions to apply to clients in window data sources, regardless of ROI.
    # For example a user with can_view_clients and can_view_name can view the names of all clients in window data sources
```

Reply:
Maybe this should be written from the alternate perspective.  Something like:
In the legacy role-based context, visibility of client attributes is global if the user can access a client within the "search" or "view" contexts for data sources that are "visible in the window."  This means we do not need to check ROI to determine if the user can see a client's name as long as they have a role that provides that permission AND can access the client through a window data source and `can_view_clients` or `can_search_all_clients` permission.

(That's probably not any more clear, but maybe?)